### PR TITLE
Limit search pattern for remove command

### PR DESCRIPTION
**Pull request by j-mracek**
It prevents additional removal after succesful remove using the same spec. Spec `foo`. The original code during the first run removes (dnf5 remove foo) all packages with the name `foo`. In the next run (dnf5 remove foo) it removes all packages providing `foo`. The next run will remove packages providing binary `foo`. This behavior is inconsistent.

Resolves: https://issues.redhat.com/browse/RHEL-5747

CI-tests - https://github.com/rpm-software-management/ci-dnf-stack/pull/1430

### CODE REVIEW

File: doc/commands/repoquery.8.rst [hunk 1]
```diff
@@ -32,8 +32,8 @@ Description
 ===========
 
 The ``repoquery`` command in ``DNF5`` is used for querying packages by matching
-various input criteria from the user. Arguments defined in ``spec`` list could be
-either file paths or package specs.
+various input criteria from the user. rguments defined in ``spec`` list is used
```
Comment (ppisar) [inline doc/commands/repoquery.8.rst:35]: "rgument" is a typo. Regarding "is used", did you mean "An argument"?

**Comment by ppisar** [comment]
Could you make a commit subject more specific? Like "Limit search pattern for remove command to NEVRAs and files"

**Comment by ppisar** [comment]
This is a quite serious change which deserves a test. Do you plan enhancing ci-dnf-stack?
I noticed that testing-farm tests fail recently. It would be great to fix them before merging this patch.

**Comment by j-mracek** [comment]
> This is a quite serious change which deserves a test. Do you plan enhancing ci-dnf-stack? I noticed that testing-farm tests fail recently. It would be great to fix them before merging this patch.

The code require following change of CI - https://github.com/rpm-software-management/ci-dnf-stack/pull/1430. It removes all tests for formal behavior and add one test for the new behavior.

**Comment by j-mracek** [comment]
@ppisar Done

File: doc/commands/remove.8.rst [hunk 2]
```diff
@@ -25,13 +25,15 @@
...
 
-``dnf5 remove [options] <package-spec>...``
+``dnf5 remove [options] [<spec>...]``
 
 
 Description
 ===========
 
 The ``remove`` command in ``DNF5`` is used for removing installed packages from the system.
+Arguments defined in ``spec`` list is used as ``<package-file-spec>``.
```
Comment (ppisar) [inline doc/commands/remove.8.rst:35]: A grammar error in "Arguments... is". Correct it to "Arguments... are".

**Comment by j-mracek** [comment]
@ppisar Thank you for couching  it. Fixed.

**Pull request merged**